### PR TITLE
Some performance improvements for big environments

### PR DESCRIPTION
**Pull request by ghost**
- Change from `System.Array` to `System.Collections.ArrayList`
- Fix `$Count = $null `($count is never null)
- Filter SecurityEnabled groups instead of `Where-Object`

### CODE REVIEW

File: AzureHound.ps1 [hunk 1]
```diff
@@ -34,19 +34,22 @@ function Get-AzureGraphToken
...
     
     Write-Host "Writing output for $($Type)"
-    $Count = $Coll.Count
-    if ($Count = $null) {
-        $Coll = @($Coll)
+	if ($null -eq $Coll) {
+        $Coll = New-Object System.Collections.ArrayList
+		$Count = 0
     }
-    $Output = New-Object PSObject
```
Comment (DPeltier) [inline AzureHound.ps1:49]: Review following an issue. Removing this line causes an error

Comment (DPeltier) [reply AzureHound.ps1:49]: This line ( $Output = New-Object PSObject) shoudn't be removed

File: AzureHound.ps1 [hunk 2]
```diff
@@ -34,19 +34,22 @@ function Get-AzureGraphToken
...
+	else {
+		$Count = $Coll.Count
```
Comment (DPeltier) [inline AzureHound.ps1]: $Count = $Coll.Count can be defined before the if ($null -eq $Coll) {

**Comment by andyrobbins** [comment]
Thank you, Bastien, this is a FANTASTIC PR. @hausec tested this in a fairly large tenant and it took the total run time down from 3 hours 42 minutes to 36 minutes.

**Pull request merged**